jerry-core/debugger/jerry-debugger.c

```diff
@@ -1 +1,113 @@
-//TODO: Implement the socket communication
+/* Copyright 2016 University of Szeged.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <errno.h>
+#include <arpa/inet.h>
+#include <unistd.h>
+#include <string.h>
+#include <stdbool.h>
+
+#include "jerry-debugger.h"
+#include "jerry-port.h"
+
+#define PORT 5001
+#define BACKLOG 1
+
+int sock;                     /**< socket file descriptor for the remote communication */
+int connected;                /**< hold the file descriptor for the accepted socket */
```

Comment:
These should be static. Perhaps they could even be part of the global context but I am not sure.